File: 2ndweekchallenges/1_7_2025challenges/Problem.py

num=3
t=0
c={}
for i in range(1,num+1):
    a=i 
    n=0
    while(a>0):
        if a in c:
            t+=n 
            t+=c[a] 
            break
        a=a&a-1
        n+=1  
    print('n',n,'t',t,'i',i)
    print(a)
    if i not in c:
        print('hi',i)
        c[i]=n 
    t+=n 

print(t)

File: 2ndweekchallenges/2-7_2025challenges/Problem.java
import java.util.HashMap;

public class Problem {
    public static void main(String[] args) {
        int num=100;
        int totalnoofones=0;
        System.out.println("hi");
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int i=1;i<=num;i++){
            int nofones=0;
            System.out.println("1");
            int a=i;
            while(a>0){
                System.out.println("helo");
                if(map.containsKey(a)){
                    totalnoofones+=map.get(a);
                    totalnoofones+=nofones;
                    break;
                }
                a=a&(a-1);
                nofones+=1;
                
                
            }
            map.put(a, nofones);
            totalnoofones+=nofones;
        }
        System.out.println(totalnoofones);
        map.forEach((s,a)->System.out.println(s+"  "+a));
    }
    
}
